mxcube3/routes/Login.py

```diff
@@ -113,22 +132,112 @@ def signout():
     """
     Signout from Mxcube3 and reset the session
     """
+    if mxcube.CURRENTLY_MOUNTED_SAMPLE:
+        if mxcube.CURRENTLY_MOUNTED_SAMPLE.get('location', '') == 'Manual':
+            mxcube.CURRENTLY_MOUNTED_SAMPLE = ''
+
+    user = remove_user(session.sid)
+
+    if not logged_in_users(exclude_inhouse=False):
+        qutils.save_queue(session)
+        mxcube.queue = qutils.new_queue()
+        mxcube.shapes.clear_all()
+        qutils.init_queue_settings()
+        mxcube.SELECTED_PROPOSAL = None
+        mxcube.SELECTED_PROPOSAL_ID = None
+        clear_messages()
+        # ony clear hwobj if this was the last user
+        if hasattr(mxcube.session, 'clear_session'):
+            mxcube.session.clear_session()
+    session.clear()
+
+    logging.getLogger('HWR').info('[Login] %s user signout' % user['loginID'])
+
+    return make_response("", 200)
+
+@mxcube.route("/mxcube/api/v0.1/forcesignout")
+def forcesignout():
+    """
+    Force signout from Mxcube3 and reset the session
+    """
+    global LOGGED_IN_USER
+
     qutils.save_queue(session)
     mxcube.queue = qutils.new_queue()
     mxcube.shapes.clear_all()
-    qutils.init_queue_settings()
-    if hasattr(mxcube.session, 'clear_session'):
-        mxcube.session.clear_session()
+    mxcube.session.clear_session()
 
     if mxcube.CURRENTLY_MOUNTED_SAMPLE:
         if mxcube.CURRENTLY_MOUNTED_SAMPLE.get('location', '') == 'Manual':
             mxcube.CURRENTLY_MOUNTED_SAMPLE = ''
 
-    user = remove_user(session.sid)
-    session.clear()
+    LOGGED_IN_USER = None
+    state_storage.flush()
+
+    mxcube.SELECTED_PROPOSAL = None
+    mxcube.SELECTED_PROPOSAL_ID = None
 
+    session.clear()
+    socketio.emit("signout", {}, namespace='/hwr')
     return make_response("", 200)
 
+@mxcube.route("/mxcube/api/v0.1/forceusersignout", methods=["POST"])
+def forceusersignout():
+    """
+    Force signout of a given user from Mxcube3
+    """
+    logging.getLogger("HWR").info('Forcing signout of user')
+    user_id = request.get_json()['sid']
+    user = get_user_by_sid(user_id)
+    # try by name
+    if user is None:
+        user = get_user_by_name(user_id)
+    remove_user(user.get('sid'))
+    socketio.emit("signout", user, room=user["socketio_sid"], namespace='/hwr')
+    logging.getLogger('HWR').info('[Login] %s user forced signout' % user['loginID'])
+
+    users = get_users()
+    
+    if len(users) == 0:
+        # no one else is connected
+        # we are here because the user force logut itself
+        mxcube.queue = qutils.new_queue()
+        mxcube.shapes.clear_all()
+        mxcube.session.clear_session()
+
+        if mxcube.CURRENTLY_MOUNTED_SAMPLE:
+            if mxcube.CURRENTLY_MOUNTED_SAMPLE.get('location', '') == 'Manual':
+                mxcube.CURRENTLY_MOUNTED_SAMPLE = ''
+
+        LOGGED_IN_USER = None
```

Comment:
Is this somekind of gobal or is it just a local variable with caps ?

Reply:
it's a global, but I think is dead code... need to check